Licensing: add Migration guide + some fixes (#8843)
DevExpress/devextreme-documentation: concepts/Common/Licensing/11 Migrate to the New Licensing Mechanism.md

If your application already follows the setup described in [Validate Your License Key](/concepts/Common/Licensing/10%20Validate%20Your%20License%20Key.md '/Documentation/Guide/Common/Licensing/#Validate_Your_License_Key'), you do not need to change your application code.

### Step 1: Register Your DevExpress License Key

Register your DevExpress license key on your machine using one of the methods described in [License Key for DevExpress Products](https://docs.devexpress.com/GeneralInformation/405494/trial-register/set-up-your-dev-express-license-key#register-your-key-basics). If you already use other DevExpress products, the key is likely already installed. Update the key as you normally would when upgrading to a new major version.

### Step 2: Verify Your License Key File

Your application most likely already contains a file that stores the DevExtreme runtime key. The runtime key file is typically excluded from version control using `.gitignore`. DevExtreme recommends the following file names and locations:

- `src/devextreme-license.ts` — for modular applications

> - `src/devextreme-license.js` — for non-modular (script-based) applications

Reviewer: (minor)
 (script-based) -> (`<script>`-based)

> ### Step 3: Add a `prebuild` Script

Reviewer: Modular and non modular approaches should be shown for all our frameworks (like it used to be)
Example:

### Step 3: Generate runtime key
The `devextreme-license` CLI reads your registered DevExpress key and generates a DevExtreme-compatible runtime key. 

**Modular applications**
Add the following script to your `package.json` to automatically regenerate the runtime key file each build:
```json
{
      "scripts": {
        "prebuild": "devextreme-license --out src/devextreme-license.ts --force"
      }
    }
```

The resulting file fully matches the format we [recommended](https://js.devexpress.com/Documentation/Guide/Common/Licensing/#Validate_Your_License_Key/Manual_Configuration/Modular) in the documentation, but it contains the key in the new format.
 
**jQuery / Non-modular applications**
Run `devextreme-license` with the `--non-modular`.
```
devextreme-license --non-modular --out src/devextreme-license.js --force
```
The resulting file fully matches the format we [recommended](https://js.devexpress.com/Documentation/Guide/Common/Licensing/#Validate_Your_License_Key/Manual_Configuration/Non-Modular) in the documentation, but it contains the key in the new format.

Verify that the generated file is included using a `<script>` tag:
```
<script src="./devextreme-license.js" type="text/javascript"></script>
```

### Step 4: Verify Your Application
In most cases, no application code changes are required. Verify that the runtime key is read from the correct file in the location where you pass it to `DevExpress.config(GlobalConfig)`.

If you previously used your license key from a custom file, use `--out` parameter to specify custom file path or run `devextreme-license` without parameters (it outputs the key directly to the CLI).

> Add the following script to your `package.json` to automatically regenerate the runtime key file before each build (update the `--out` path/extension to match your runtime key file):

Reviewer: (minor)
we can say 'before each build' -> 'each build'
because prebuilding is a part of building

    {
      "scripts": {
        "prebuild": "devextreme-license --out src/devextreme-license.ts --force"
      }
    }

> The `devextreme-license` CLI reads your registered DevExpress key and generates a DevExtreme-compatible runtime key. Pass the runtime key to the [DevExpress.config(GlobalConfig)](/api-reference/50%20Common/utils/config().md '/Documentation/ApiReference/Common/Utils/#config') method.

Reviewer: Pass the runtime key to the ... -> Check that the new generated runtime key is passed to the ....

Because previously they are supposed to pass an old one. 

UPD: we have the following step about config(), we can skip it here

To preview the generated key without writing a file, run `devextreme-license` without parameters (it outputs the key directly to the CLI).

---

> ##### jQuery

Reviewer: jQuery / Non-modular applications

> DevExtreme v26.1 introduces a new licensing mechanism. Follow the steps below to migrate your application when you upgrade from an earlier version.

Reviewer 2: ```suggestion
DevExtreme v26.1 introduces a new licensing mechanism. Follow the steps below to migrate your application from an earlier version.
```

Applications that already follow the setup described in [Validate Your License Key](/concepts/Common/Licensing/10%20Validate%20Your%20License%20Key.md '/Documentation/Guide/Common/Licensing/#Validate_Your_License_Key') do not require application code changes.

### Step 1: Register Your DevExpress License Key

> Register your DevExpress license key on your machine using one of the methods described in [License Key for DevExpress Products](https://docs.devexpress.com/GeneralInformation/405494/trial-register/set-up-your-dev-express-license-key#register-your-key-basics). If you already use other DevExpress products, the key is likely already installed. Update the key as you normally would when upgrading to a new major version.

Reviewer 2: ```suggestion
Register your DevExpress license key on your machine using one of the methods described in [License Key for DevExpress Products](https://docs.devexpress.com/GeneralInformation/405494/trial-register/set-up-your-dev-express-license-key#register-your-key-basics). If you already use other DevExpress products, the key is likely already installed. Update the key as usual when you upgrade to a new major version.
```

##### jQuery / Non-Modular Applications

Run `devextreme-license` with the `--non-modular` flag:

    devextreme-license --non-modular --out src/devextreme-license.js --force

The resulting file fully matches the format we [recommend](https://docs.devexpress.com/GeneralInformation/405494/trial-register/set-up-your-dev-express-license-key#option-2-non-modular-applications-jquery-cdn-based) in the documentation, but contains the key in the new format.

Verify that the generated file is included using a `<script>` tag:

    <script src="./src/devextreme-license.js" type="text/javascript"></script>

### Step 3: Verify Your Application

In most cases, no application code changes are required. Verify that the runtime key is read from the correct file in the location where you pass it to [DevExpress.config(GlobalConfig)](/api-reference/50%20Common/utils/config().md '/Documentation/ApiReference/Common/Utils/#config').

If you previously used your license key from a custom file, use the `--out` parameter to specify a custom file path, or run `devextreme-license` without parameters (it outputs the key directly to the CLI).

### Step 4: Configure CI/CD

> For CI/CD environments, the `prebuild` script added in Step 2 is sufficient on the build side. Additionally, set your DevExpress license key in an environment variable on your build server. For full instructions, refer to [Deployment and Continuous Integration (CI)](/concepts/Common/Licensing/13%20Deployment%20and%20Continuous%20Integration%20(CI).md '/Documentation/Guide/Common/Licensing/#Deployment_and_Continuous_Integration_CI').

Reviewer 2: ```suggestion
For CI/CD environments, the `prebuild` script added in Step 2 is sufficient on the build side, but you must also set your DevExpress license key in an environment variable on the build server. For complete instructions, see [Deployment and Continuous Integration (CI)](/concepts/Common/Licensing/13%20Deployment%20and%20Continuous%20Integration%20(CI).md '/Documentation/Guide/Common/Licensing/#Deployment_and_Continuous_Integration_CI').
```

DevExtreme v26.1 introduces a new licensing mechanism. If you are upgrading from an earlier version, follow the steps below to migrate your application.

If your application already follows the setup described in [Validate Your License Key](/concepts/Common/Licensing/10%20Validate%20Your%20License%20Key.md '/Documentation/Guide/Common/Licensing/#Validate_Your_License_Key'), you do not need to change your application code.

### Step 1: Register Your DevExpress License Key

Register your DevExpress license key on your machine using one of the methods described in [License Key for DevExpress Products](https://docs.devexpress.com/GeneralInformation/405494/trial-register/set-up-your-dev-express-license-key#register-your-key-basics). If you already use other DevExpress products, the key is likely already installed. Update the key as you normally would when upgrading to a new major version.

> ### Step 2: Verify Your License Key File

Reviewer: We should merge step 2 and step 3. Because now step 2 isn't an independent step

Your application most likely already contains a file that stores the DevExtreme runtime key. The runtime key file is typically excluded from version control using `.gitignore`. DevExtreme recommends the following file names and locations:

- `src/devextreme-license.ts` — for modular applications
- `src/devextreme-license.js` — for non-modular (`<script>`-based) applications

### Step 3: Generate Runtime Key

The `devextreme-license` CLI reads your registered DevExpress key and generates a DevExtreme-compatible runtime key.

##### Modular Applications

Add the following script to your `package.json` to automatically regenerate the runtime key file each build (update the `--out` path/extension to match your runtime key file):

    {
      "scripts": {
        "prebuild": "devextreme-license --out src/devextreme-license.ts --force"
      }
    }

> The resulting file fully matches the format we [recommended](/concepts/Common/Licensing/10%20Validate%20Your%20License%20Key.md '/Documentation/Guide/Common/Licensing/#Validate_Your_License_Key') in the documentation, but contains the key in the new format.

Reviewer: I'm not sure about these links 
I expect that we link here to 25.2 topic and in particular to

MODULAR
<img width="1696" height="736" alt="Image" src="https://github.com/user-attachments/assets/a1327bc8-f0c6-4425-821a-db94ce6952f8" />

NON Modular

<img width="1460" height="560" alt="Image" src="https://github.com/user-attachments/assets/f1edd141-988d-4243-95e7-715c48848913" />
